src/graphics/image.rs

```diff
@@ -186,10 +189,31 @@ impl Image {
     /// This function doesn't check the validity of the pixel
     /// coordinates, using out-of-range values will result in
     /// an undefined behaviour.
-    pub unsafe fn set_pixel(&mut self, x: u32, y: u32, color: Color) {
+    pub unsafe fn set_pixel_unchecked(&mut self, x: u32, y: u32, color: Color) {
         ffi::sfImage_setPixel(self.image, x, y, color)
     }
 
+    /// Change the color of a pixel in an image
+    ///
+    /// # Arguments
+    /// * x - X coordinate of pixel to change
+    /// * y - Y coordinate of pixel to change
+    /// * color - New color of the pixel
+    pub fn set_pixel(&mut self, x: u32, y: u32, color: Color) -> Result<(), SetPixelError> {
+        let image_size = self.size();
+        if x >= image_size.x {
+            return Err(SetPixelError::XTooLarge(x, image_size.x - 1));
+        }
+        if y >= image_size.y {
+            return Err(SetPixelError::YTooLarge(y, image_size.y - 1));
+        }
+
+        // Since we check for index validity before setting the pixel, it is safe unless the
+        // image texture has been unloaded, but I doubt you can even do that.
```

Comment:
Using the word "texture" here is confusing, since this is an image in system memory, not a texture in video memory.

Reply:
Removing the word "texture" is enough here. Fixed!